pkg/processor/runtime/rpc/connection/abstract.go

```diff
@@ -0,0 +1,362 @@
+/*
+Copyright 2024 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package connection
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"io"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/common/status"
+	"github.com/nuclio/nuclio/pkg/processor/controlcommunication"
+	"github.com/nuclio/nuclio/pkg/processor/runtime"
+	"github.com/nuclio/nuclio/pkg/processor/runtime/rpc/controlmessagebroker"
+	"github.com/nuclio/nuclio/pkg/processor/runtime/rpc/encoder"
+	"github.com/nuclio/nuclio/pkg/processor/runtime/rpc/result"
+
+	"github.com/nuclio/errors"
+	"github.com/nuclio/logger"
+)
+
+type AbstractConnectionManager struct {
+	Logger logger.Logger
+
+	MinSocketsNum int
+	MaxSocketsNum int
```

Comment:
Isn't this only relevant to the `SocketAllocator`?

Reply:
@TomerShor it's gonna be general for every manager, I just need to rename it indeed